### PR TITLE
Chore: Add a script for testing with more control

### PR DESCRIPTION
**Pull request by fa93hws**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: Add a script to `package.json`

**What changes did you make? (Give an overview)**
Issue link: https://github.com/eslint/eslint/issues/12442
Not sure whether I am the only one, I tends to not read the documentation page by page before the development and hence I neglect the fact that there is an option to have better control on the unit testing.
Also, the documentation in https://eslint.org/docs/developer-guide/unit-tests#running-individual-tests isn't complete as the developer can utilize all options from the mocha cli such as `--watch`.
I think it might better to have it in the `package.json` as a script so that it's easier for someone else in the future to find this option easier.

**Is there anything you'd like reviewers to focus on?**
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   "main": "./lib/api.js",
   "scripts": {
     "test": "node Makefile.js test",
+    "test:cli": "./node_modules/.bin/mocha",
```
Comment (platinumazure) [inline package.json]: Would it work to simply say "mocha" here?

Comment (fa93hws) [reply package.json]: It takes me quite a while to think about the naming and I can't think of a good one.🤦‍♂️
Maybe `mocha` works better.

File: package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
   "main": "./lib/api.js",
   "scripts": {
     "test": "node Makefile.js test",
+    "mocha": "./node_modules/.bin/mocha",
```
Comment (platinumazure) [inline package.json]: My apologies, I meant to say you could get rid of the path prefix due to how `npm run` works.

```
"test:cli": "mocha"
```

Comment (fa93hws) [reply package.json]: Oh sorry, I see

Comment (platinumazure) [reply package.json]: No need to apologize- this was my fault. :smile: Thanks for your patience!

**Comment by g-plane** [comment]
Should we add `--reporter=progress`?

**Comment by fa93hws** [comment]
> Should we add `--reporter=progress`?

I am slightly toward to "No"

If the developers are split into two groups, familiar with the Mocha or not.
For those who familiar with the Mocha, there isn't much difference between having it or not.
For those who aren't familiar with it, chances that they are using it according to https://eslint.org/docs/developer-guide/unit-tests#running-individual-tests. The default `reporter` would be more suitable for few files.

But I believe you have much more context than I do so either is fine for me.

**Comment by platinumazure** [comment]
I'll go ahead and merge this as-is, without `--reporter=progress`. If we need to change this later, we can do so in a separate PR.

Thanks @fa93hws for contributing!

**Pull request merged**